### PR TITLE
Misc cleanups

### PR DESCRIPTION
**Pull request by AdrianVovk**
Some commits split out of #30840, as [requested](https://github.com/systemd/systemd/pull/30840#issuecomment-1898136383)

### CODE REVIEW

File: src/basic/fd-util.h [hunk 1]
```diff
@@ -52,6 +52,11 @@ static inline void fclosep(FILE **f) {
         safe_fclose(*f);
 }
 
+static inline void *close_fd_ptr(void *p) {
```
Comment (YHNdnzj) [inline src/basic/fd-util.h]:
```suggestion
static inline void* close_fd_ptr(void *p) {
```

(Not your fault. But if you touch this, please write T* for function return types.

**Review by YHNdnzj — APPROVED** [review]

File: src/basic/fd-util.h (hunk 1 above)
Comment (AdrianVovk) [reply src/basic/fd-util.h]: Done

**Comment by bluca** [comment]
@mrc0mmand is the `TEST-35-LOGIN` failure a known issue?

**Comment by YHNdnzj** [comment]
> is the `TEST-35-LOGIN` failure a known issue?

Yeah, that's #30910

**Pull request merged**